notebooks/byod.py

```diff
@@ -98,30 +98,39 @@ def upload_columns(table: str, columns: Dict[str, List[Any]]):
             tsv_data += os.linesep + "\t".join([f"{i}", *[columns[h][i] for h in column_headers]])
         upload_data_table(tsv_data)
 
+    def parse_cram_crai(filename: str):
+        sample, ext = filename.str.rsplit(".", 1)
+        return sample, ext
+
     def create_cram_crai_table(table: str, listing: Iterable[str]):
         crams = dict()
         crais = dict()
         for key in listing:
             _, filename = key.rsplit("/", 1)
-
-            parts = filename.split(".")
-            if 3 == len(parts):  # foo.cram.crai branch
-                sample, _, ext = parts
-            elif 2 == len(parts):  # "foo.cram" or "foo.crai" branch
-                sample, ext = parts
-            else:
-                raise ValueError(f"Unable to parse '{filename}'")
-
+            sample, ext = parse_cram_crai(filename)
             if "cram" == ext:
                 crams[sample] = key
             elif "crai" == ext:
                 crais[sample] = key
             else:
```

Comment:
This hasn't been resolved yet.

Reply:
This prevents a KeyError, doesn't it?

Reply:
The final `else` is unnecessary.  Run this with and without the commented part:

```
crams = dict()
crais = dict()

files = ['a.cram', 'a.crai', 'b.baloney', 'c.crai', 'c.cram']


def release_the_files(files):
    for file in files:
        if file.endswith('cram'):
            crams[file] = 'a'
        elif file.endswith('crai'):
            crais[file] = 'c'
        # else:
        #     continue
    print('I return!')
    return crams, crais


crams, crais = release_the_files(files)

print(crams, crais)
```

Reply:
Note, the change could also be done in a separate PR.

Reply:
Oh, I see. I always add a terminal else when I also use elif out of habit. I think my Java professor drilled it into me even though it's not required. I can remove it though.